src/io/json.cpp

```diff
@@ -588,8 +586,9 @@ namespace fc
    }
 
    template<typename T>
-   void to_stream( T& os, const variant& v, json::output_formatting format )
+   void to_stream( T& os, const variant& v, json::output_formatting format, uint32_t max_depth )
    {
+      FC_ASSERT( max_depth > 0, "Too many nested objects!" );
```

Comment:
There is an issue related to this, although not necessarily caused by it. For example, among this code:

```
object o( a_nested_object );
variant a;
try {
   to_stream( nested_object, a );
} FC_CAPTURE_AND_RETHROW ( (o) );
```

`FC_CAPTURE_AND_RETHROW ();` will try to jsonify `o` which is the nested object, so it will throw another exception.

We need to carefully check all `catch`es and make sure they don't throw again.